### ISSUE TITLE
ERROR: Cannot retrieve any response status before encountering an EPIPE or ECONNRESET exception

### ISSUE THREAD

**Issue by jamieburchell** [s3tools/s3cmd#912]
I've started noticing this new error in my backup logs since upgrading to s3cmd version 2.0.0 when using sync mode.

The backup appears to still complete succcesfully with no other errors.

**Comment by fviard** [comment]
Just after, you should have the root cause of the error in a "warning" message:
"Upload failed: ..."
Can you tell me what it is?
The previous message is a recent addition for early errors in "requests".

**Comment by jamieburchell** [comment]
That's just it, there were no failure messages afterwards - the process successfully uploaded the files it needed to. Perhaps the logging needs to be more verbose to see that? Could the error have affected getting a file list or similar instead of sending the file?

**Comment by jamieburchell** [comment]
I increased the verbosity level to warning and re-ran but did not get the error this time.

I remembered why the verbosity level was at error ... I still get the retrying failed request spewing out in to the logs despite the uploads always working.

**Comment by fviard** [comment]
you can run the command with the "-d" flag to have the most log.

**Comment by jamieburchell** [comment]
Encountered again, the next line was failed to upload error 32 broken pipe, retried then worked which is likely what happened before but with warnings turned off I wouldn't have seen the retry.

My logs are full of warnings: retrying failed request. It's the reason I turned off warnings.

**Comment by jamieburchell** [comment]
This issue is still not resolved in 2.0.2.
My logs always start with ERROR: Cannot retrieve any response status before encountering an EPIPE or ECONNRESET exception when syncing folders with large numbers of files in a folder in my bucket.

**Comment by elsmorian** [comment]
Can confirm we are also seeing this-

```
ERROR: Cannot retrieve any response status before encountering an EPIPE or ECONNRESET exception
WARNING: Upload failed: /my-bucket/my-file ([Errno 32] Broken pipe)
WARNING: Waiting 3 sec...
```

As @jamieburchell says it means running with normal output all you get is the ERROR line, where the command actually succeeds fine after a retry! 

There is also no line mentioning that the retry above was in fact successful which would be handy.

**Comment by antwal** [comment]
i have same problem with 2.1.0 with disk image copy

**Comment by fviard** [comment]
3 years, later, I think that I have finally figured it out!
See #1114 .
There is a fix in MASTER, so you if you can give it a try it would be great, otherwise it will be included in next release 2.2.0.

**Comment by jamieburchell** [comment]
Can this issue be reopened? It's still not fixed in 2.1.0...

```
ERROR: Cannot retrieve any response status before encountering an EPIPE or ECONNRESET exception
WARNING: Upload failed: /Photos/2020/62D.mov ([Errno 32] Broken pipe)
WARNING: Waiting 3 sec...
upload: '/share/Multimedia/Photos/2020/62D.mov' -> 's3://peanet/Photos/2020/62D.mov' (1793027 bytes in 2.3 seconds, 768.38 KB/s) [1 of 59]
```

**Comment by jamieburchell** [comment]
@fviard Happy to confirm this warning is no longer present with the changes in master. Thanks for your efforts!
